packages/fcl/src/wallet-provider-spec/draft-v3.md

```diff
@@ -0,0 +1,569 @@
+
+FCL acts in many ways as a protocol to facilitate communication and configuration between the different parties involved in a blockchain application. An _Application_ can use FCL to _authenticate_ users, and request _authorizations_ for transactions, as well as mutate and query the _Blockchain_. An application using FCL offers it's _Users_ a way to connect and select any number of Wallets Providers and their Wallet Services. A selected _Wallet_ provides an Applications instance of FCL with configuration information about itself and its Wallet Services, allowing the _User_ and _Application_ to interact with them.
+
+In the following paragraphs we'll explore ways in which you, as a wallet developer, can integrate with FCL by providing implementataions of various FCL services. 
+
+The following services will be covered:
+- Authentication (Authn) Service
+- Authorization (Authz) Service
+- User Signature Service
+- Pre-Authz Service
+
+# Service Methods
+
+FCL Services are your way as a Wallet Provider of configuring FCL. In accomplishing these service, FCL employs various Service Methods. Services Methods are the ways FCL can talk to your wallet. You as a Wallet Provider get to decide which of these service methods each of your services use to communicate with you.
+
+Sometimes services just configure FCL and thats it. An example of this case can be seen with the Authentication Service and the OpenID Service.
+With those two services you are simply telling FCL "here is a bunch of info about the current user". (You will see that those two services both have a `method: "DATA"` field in them.
+Currently these are the only two cases that can be a data service.)
+
+Other services can be a little more complex. For example, they might require a back and forth communication between FCL and the Service in question.
+Ultimately we want to do this back and forth via a secure back-channel (https requests to servers), **but in some situations that isn't a viable option, so there is also a front-channel option**.
+Where possible, you should aim to provide a back-channel support for services, and only fall back to a front-channel if absolutely necessary.
+
+Back-channel communications use `method: "HTTP/POST"`, while front-channel communications use `method: "IFRAME/RPC"`, `method: "POP/RPC"` or `method: "TAB/RPC`.
+
+| Service Method | Front  |  Back |
+|----------------|--------|-------|
+| HTTP/POST      |   ⛔   |   ✅   |
+| IFRAME/RPC     |   ✅   |   ⛔   |
+| POP/RPC        |   ✅   |   ⛔   |
+| TAB/RPC        |   ✅   |   ⛔   |
+
+It's important to note that regardless of the method of communication, the data that is sent back and forth between the parties involved is the same.
+
+### IFRAME/RPC
```

Comment:
Can we maybe add the fact that its considered a font channel here too, just to double down.
maybe something like
```
### IFRAME/RPC (Front Channel)
```